src/middleware/validate/validateAddressMiddleware.ts

```diff
@@ -35,11 +35,21 @@ function checkAddress(address: string): [boolean, string | undefined] {
 	try {
 		decoded = base58Decode(address);
 	} catch (error) {
+		// If ss58 doesn't pass, we try H160 address.
+		if (address.startsWith('0x')) {
+			decoded = Uint8Array.from(Buffer.from(address.slice(2), 'hex'));
+
+			if (decoded.length != 20) {
+				return [false, `Invalid decoded address length ${decoded.length} for H160 account`];
+			}
```

Comment:
There is `isEthereumAddress` that seems to do this job (and also runs more checks with `isEthereumChecksum`). Does it not work here?